elbepack/initvmaction.py

```diff
@@ -233,7 +325,51 @@ class StopAction(InitVMAction):
     def __init__(self, node):
         InitVMAction.__init__(self, node)
 
-    def execute(self, _initvmdir, _opt, _args):
+    def _stop_qemu_vm(self, initvmdir):
+        """
+        Stop the QEMU initvm.
+
+        This method tries to stop the QEMU initvm by sending a poweroff event
+        using QEMU monitor.
+        """
+        
+        if initvmdir == None:
+            if os.path.isdir('./initvm'):
+                print('Using default initvm directory "./initvm".')
+                initvmdir = './initvm'
+            else:
+                print('No initvm found!')    
+                sys.exit(207)
+
+        socket_path = os.path.join(initvmdir, 'qemu-monitor-socket')
+
+        # test if QEMU monitor unix-socket file exists, and error exit if not
+        if not os.path.exists(os.path.join(initvmdir, 'qemu-monitor-socket')):
+            print('No unix socket found for this vm!\nunable to shutdown this vm.')
+            sys.exit(212)
+
+        try:
+            with socket.socket(socket.AF_UNIX, socket.SOCK_STREAM) as client:
```

Comment:
Same as for the other socket usage above.

Reply:
socket.create_connection only works for TCP sockets